static/app/components/compactSelect/list.tsx

```diff
@@ -353,7 +353,9 @@ function List<Value extends SelectKey>({
           listState={listState}
           sizeLimitMessage={sizeLimitMessage}
           keyDownHandler={keyDownHandler}
-        />
+        >
+          {props.children as React.ReactNode}
```

Comment:
What do you think about surrounding props with PropsWithChildren type and avoid force casting?

Reply:
In this case children seems to be `CollectionChildren` from `@react-types`. It can be a render function which errors in react 18 
the type
![image](https://github.com/getsentry/sentry/assets/1400464/ce358e6e-aaab-43cf-aeef-511e92b60574)

react 18 does not seem to like passing a function to render without calling it
![image](https://github.com/getsentry/sentry/assets/1400464/73d301f3-329e-40c8-8c51-b7fa33ef5d5d)

Reply:
swapped it to using `CollectionChildren` in the sub components, that feels a bit better